### ISSUE TITLE
Ignoring broken lockfile in a CI env 

### ISSUE THREAD

**Issue by CYriuk** [pnpm/pnpm#6084]
### pnpm version:
7.26.1

### Code to reproduce the issue:
A Renovate bot MR has generated an upgrade for the [ember-cli-htmlbars
](https://www.npmjs.com/package/ember-cli-htmlbars) lib from `6.1.1` to `6.2.0` without propagating the changes in the lockfile to keep it synchronized

### Expected behavior:
Without specifying the --frozen-lockfile it should not ignore the broken lockfile and should throw an error instead

### Actual behavior:
Running `pnpm install` in a Gitlab CI environment does not apply a default --frozen-lockfile when a lockfile is present

### Additional information:

 - `node -v` prints: v18
 - Windows, macOS, or Linux?:
Linux

**Comment by cdaringe** [comment]
@zkochan, can we send a PR or option for this? We really want to make sure our lockfiles _are not corrupted_. A member of our team got 🟢 in CI, but corrupted the lockfile, which we consider a paramount failure.

**Comment by ericbiewener** [comment]
Seems like it is _supposed_ to work this way already? https://github.com/pnpm/pnpm/issues/1994

**Comment by cdaringe** [comment]
nah, that's different. this is much more focused. "the lockfile is broken, but never do i want to allow the lockfile to exist in broken state. if you find it broken, fail rather than ignore"

**Comment by ericbiewener** [comment]
I see what you're saying, but my expectation is that `--frozen-lockfile` should already fail the install in such a case. From the docs:

> If true, pnpm doesn't generate a lockfile and fails to install if the lockfile is out of sync with the manifest / an update is needed or no lockfile is present.

I would expect a broken lockfile to be considered "out of sync".

**Comment by cdaringe** [comment]
a lockfile was corrupted on checkin, no .npmrc configs are set (--frozen-lockfile or otherwise), and pnpm is _ok_ with it right now but it should not be. that's my problem statement.

are you implicitly suggesting that i should be using frozen-lockfile?

from my perspective, that's just a whole different discussion.

**Comment by ericbiewener** [comment]
Nope, not suggesting that. From my reading of the docs, `--frozen-lockfile` should be _automatically_ set when in CI, no need to explicitly pass that flag ourselves. Therefore, I would expect `pnpm install` to fail in CI when there is a broken lockfile because the lockfile should be considered "out of sync".

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1 with CI=true: an unparsable pnpm-lock.yaml fails the install with exit code 1 and the parse error, instead of being silently ignored.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
